pkg/prometheus/promcfg.go

```diff
@@ -275,10 +275,10 @@ func (cg *configGenerator) generateConfig(
 		}
 	}
 
-	for _, identifier := range probeIdentifiers {
+	for i, identifier := range probeIdentifiers {
 		scrapeConfigs = append(scrapeConfigs,
 			cg.generateProbeConfig(
-				version,
+				version, i,
```

Comment:
probes can't actually produce multiple scrape configs, so it's sufficient to just use the identifier without an integer here